spec/controllers/challenge_claims_controller_spec.rb

```diff
@@ -61,10 +74,49 @@
     end
   end
 
-  describe 'destory' do
-    context 'with a claim' do
-      it 'on an exception gives an error and redirects' do
-        fake_login_known_user(@user)
+  describe "destroy" do
+    context "for a prompt meme" do
+      let(:pm_signup) { create(:prompt_meme_signup) }
+      let(:pm_collection) { pm_signup.collection }
+
+      before do
+        allow_any_instance_of(ChallengeClaim).to receive(:user_allowed_to_destroy?).and_return(true)
```

Comment:
Does `fake_login_known_user(user)` work, instead of us having to stub this method? Same for the `claiming_user` stub below.

Reply:
I don't know - I didn't want to go to the trouble of setting up the full association tree, the way we'd do in an integration spec.

Work practice has got me used to a lot of stubbing and mocking. Work test suites tend to be very fast, and it's a breeze to test something out and get (automated) feedback quickly, compared to the 2 hour slog of our gigantic test suite. Asking around, devs told me that stubbing/mocking is a huge factor in the speed and memory performance of the test suite. I badly want some of that for ourselves!

OTOH, at work we know that all the stuff that is being mocked is being thoroughly tested in its own unit tests.

Is your concern about the stuff being mocked that it might not have enough coverage elsewhere? Or is it about mocking / stubbing not being appropriate for this scenario?

Reply:
A little of both, yeah.

- I expect this spec to test whether the controller reacts appropriately to the current logged in user with the given claim, so I would prefer to actually run `user_allowed_to_destroy?` instead of stubbing it out.
- We find gaps in our tests all the time, I'd rather err on the side of having too many tests. ChallengeClaim in particular has almost no unit tests of its own, we might as well test it here.

The whole test suite takes very long, but controller specs are decently fast to iterate on individually.

I agree a faster test suite would be amazing though. Many of our Cucumber tests don't actually need to be Cucumber tests and are better as specs, e.g. https://github.com/otwcode/otwarchive/blob/6a812486f6aaa7d20cbc7bc4fac3cbdcdcaea84a/features/other_b/errors.feature#L5.

Reply:
An issue for those tests: https://otwarchive.atlassian.net/browse/AO3-6005